File: Mod06/Labfiles/Solution/VB/ClassLab_VB/ClassLab_VB/Course.vb

Public Class Course

    Private _title As String
    Private _creditHours As Integer
    Private _program As String
    Private _instructor As String

    Public Property Title As String
        Get
            Return Me._title
        End Get
        Set(value As String)
            Me._title = value
        End Set
    End Property

    Public Property CreditHours As Integer
        Get
            Return Me._creditHours
        End Get
        Set(value As Integer)
            Me._creditHours = value
        End Set
    End Property

    Public Property Program As String
        Get
            Return Me._program
        End Get
        Set(value As String)
            Me._program = value
        End Set
    End Property

    Public Property Instructor As String
        Get
            Return Me._instructor
        End Get
        Set(value As String)
            Me._instructor = value
        End Set
    End Property

    Public Sub New()

    End Sub

    Public Sub New(title As String, credits As Integer, prog As String, inst As String)

        Me._title = title
        Me._creditHours = credits
        Me._program = prog
        Me._instructor = inst

    End Sub

    Public Function GetTitle() As String

        Return Me._title.ToUpper()

    End Function

End Class

File: Mod06/Labfiles/Solution/CS/ClassLab_CS/ClassLab_CS/Course.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassLab_CS
{
    class Course
    {
        private string _title;
        private int _creditHours;
        private string _program;
        private string _instructor;

        public string Title
        {
            get { return this._title; }
            set { this._title = value; }
        }

        public int CreditHours
        {
            get { return this._creditHours; }
            set {
                if (value >= 3 || value <= 6)
                {
                    this._creditHours = value;
                }
            }
        }

        public string Program
        {
            get { return this._program; }
            set { this._program = value; }
        }
        
        public string Instructor
        {
            get { return this._instructor; }
            set { this._instructor = value; }
        }

        public Course()
        {
            
        }
        public Course(string title, int credits, string prog, string inst)
        {
            this._title = title;
            this._creditHours = credits;
            this._program = prog;
            this._instructor = inst;
        }

        public string GetTitle()
        {
            return this._title.ToUpper();
        }

    }
}
